src/main/java/io/supertokens/webserver/RequestStats.java

```diff
@@ -97,6 +97,9 @@ public static RequestStats getInstance(Main main, AppIdentifier appIdentifier) t
             return (RequestStats) main.getResourceDistributor()
                     .getResource(appIdentifier.getAsPublicTenantIdentifier(), RESOURCE_KEY);
         } catch (TenantOrAppNotFoundException e) {
+            // appIdentifier parameter is coming from the API request and hence we need to check if the app exists
+            // before creating a resource for it. The other resources are created during init or while refreshing
```

Comment:
```suggestion
            // before creating a resource for it, otherwise someone could fill up memory by making requests for apps that don't exist. The other resources are created during init or while refreshing
```